pwa/src/components/translations/translationForm.tsx

```diff
@@ -2,96 +2,95 @@ import * as React from "react";
+    setShowSpinner(true);
+    API.Translation.getOne(id)
+      .then((res) => {
+        console.log(res.data);
+        setTranslation(res.data)
+      })
+      .catch((err) => { throw new Error('GET translation error: ' + err) })
+      .finally(() => { setShowSpinner(false) });
+  };
 
   const saveTranslation = (event) => {
     event.preventDefault();
     setLoadingOverlay(true);
 
     let body = {
-      translationTable: id,
+      translationTable: event.target.translationTable ? event.target.translationTable.value : null,
       language: event.target.language ? event.target.language.value : null,
       translateFrom: event.target.translateFrom ? event.target.translateFrom.value : null,
       translateTo: event.target.translateTo ? event.target.translateTo.value : null,
     };
 
-    let url = `${context.adminUrl}/translations`;
-    let method = "POST";
+    if (tableName && tableName !== 'new') {
+      body.translationTable = tableName
+    }
 
-    fetch(url, {
-      method: method,
-      credentials: "include",
-      headers: {"Content-Type": "application/json", 'Authorization': 'Bearer ' + sessionStorage.getItem('jwt')},
-      body: JSON.stringify(body),
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        setTranslation(data);
-        method === 'POST' && navigate("/translations")
-      })
-      .catch((error) => {
-        console.error(error);
-        setAlert(null);
-        setAlert({type: 'danger', message: error.message});
-      })
-      .finally(() => {
-        setLoadingOverlay(false);
-      })
+    if (!id) {
+      API.Translation.create(body)
+        .then((res) => { setTranslation(res.data); console.log(res) })
+        .catch((err) => { throw new Error('GET translation error: ' + err) })
+        .finally(() => {
+          setShowSpinner(false);
+          setLoadingOverlay(false);
+        });
+    }
+
+    if (id) {
+      API.Translation.update(body, id)
+        .then((res) => { setTranslation(res.data); console.log(res) })
+        .catch((err) => { throw new Error('GET translation error: ' + err) })
+        .finally(() => {
+          setShowSpinner(false);
+          setLoadingOverlay(false);
+        });
+    }
   }
 
   return (
     <>
-      {
-        alert !== null &&
-        <FlashMessage duration={5000}>
-          <Alert alertClass={alert.type} body={function () {
-            return (<>{alert.message}</>)
-          }}/>
-        </FlashMessage>
-      }
       <form id="dataForm" onSubmit={saveTranslation}>
         <Card
           title={title}
           cardHeader={function () {
             return (
               <div>
-                <Link className="utrecht-link" to={"/translations"}>
+                <Link className="utrecht-link" to={translation?.translationTable ? `/translation-tables/${translation.translationTable}` : (tableName && tableName !== 'new') ? `/translation-tables/${tableName}` : '/translation-tables'}>
```

Comment:
By separating the two forms, we wouldn't need complex (and very hard to manage and build upon) checks like these.